### PR TITLE
LPS-186042 | Add Autom. Test CanBeSearchedByTitle

### PR DESCRIPTION
**Pull request by eltonccdantas**
[Ticket](https://liferay.atlassian.net/browse/LPS-186037)

Given the Feature Flags page
//Control Panel>Instance Settings
When the administrator searches for a feature flag
//click the search field and type an existing FF
Then assert the desired feature flag is displayed

### CODE REVIEW

**Comment by eltonccdantas** [comment]
ci:help

**Comment by eltonccdantas** [comment]
ci:test:relevant

**Comment by eltonccdantas** [comment]
ci:stop:relevant

**Comment by eltonccdantas** [comment]
ci:test:relevant

**Comment by eltonccdantas** [comment]
ci:stop:relevant

File: modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlag.testcase [hunk 1]
```diff
@@ -90,4 +94,27 @@ definition {
...
+
+		task ("When the administrator searches for a feature flag") {
+			Type(
+				locator1 = "FeatureFlag#SEARCH_FIELD",
+				value1 = "Terms and Conditions pages");
+
+			KeyPress(
+				locator1 = "FeatureFlag#SEARCH_FIELD",
+				value1 = "\ENTER");
+		}
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlag.testcase:110]: Could you try replacing this with existing macro `Search.searchCP`.

**Comment by eltonccdantas** [comment]
ci:test:sf

**Comment by eltonccdantas** [comment]
ci:help

**Comment by eltonccdantas** [comment]
ci:test:relevant

File: modules/apps/feature-flag/feature-flag-test/src/testFunctional/paths/FeatureFlag.path [hunk 2]
```diff
@@ -15,6 +15,16 @@
 		<td>//*[contains(text(),'${title_row}')]//ancestor::dd//label//span[contains(@class,'toggle-switch-label')][contains(text(),'${toggle_state}')]</td>
 		<td></td>
 	</tr>
+	<tr>
+		<td>SEARCH_FIELD</td>
+		<td>//input[contains(@aria-label,'Search')]</td>
+		<td></td>
+	</tr>
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-test/src/testFunctional/paths/FeatureFlag.path:22]: This path is no longer needed. It's not used anywhere.

File: modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlag.testcase [hunk 3]
```diff
@@ -21,6 +21,10 @@ definition {
 			Click(
 				key_itemName = "Feature Flags",
 				locator1 = "ListGroupItem#ITEM_TEXT");
+
+			Click(
+				key_menuBar = "Release",
+				locator1 = "FeatureFlag#MENU_BAR");
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlag.testcase:27]: Please replace lines 16 thru 27 with PortalSettings.gotoConfiguration(). We don't need to add more paths.

**Review by john-co — CHANGES_REQUESTED** [review]
Please see comments to refactor the test.

**Comment by eltonccdantas** [comment]
ci:test:sf

**Comment by eltonccdantas** [comment]
ci:test:relevant

**Comment by eltonccdantas** [comment]
ci:test:relevant

**Comment by eltonccdantas** [comment]
ci:stop:relevant

**Comment by john-co** [comment]
hi @eltonccdantas running `ci:test:relevant` shouldn't return any results since this PR only contains poshi file changes.

**Comment by john-co** [comment]
ci:test:peds-sm:nocompile

**Review by john-co — APPROVED** [review]
please see changes

**Comment by eltonccdantas** [comment]
> please see changes

Thanks, tiny things that escaped from my eyes...

**Comment by eltonccdantas** [comment]
> hi @eltonccdantas running `ci:test:relevant` shouldn't return any results since this PR only contains poshi file changes.

Alright... I'll remember it next time.

**Comment by eltonccdantas** [comment]
ci:test:peds-sm:nocompile

**Comment by john-co** [comment]
Hi @eltonccdantas, my changes weren't saved when you force pushed...do you need me to add the commits back?

**Comment by eltonccdantas** [comment]
Hey @john-co, these tests are failing for the same reason, it seems that it cannot find `Terms and Conditions pages` 's toggle... When I reproduce the tests locally, they pass. I also tried changing the window size by adding `SetWindowSize(value1 = "1920,1540");` to the setUp phase, because maybe the scroll did not work as expected, but I got the same error... Do you know how to fix it?

File: modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlag.testcase [hunk 4]
```diff
@@ -93,7 +93,7 @@ definition {
 		}
 	}
 
-	@description = "LPS-167698 - This tests asserts that a feature flag can be searched by tittle"
+	@description = "LPS-167698 - This tests asserts that a feature flag can be searched by tittle."
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlag.testcase:96]: typo: title

**Comment by john-co** [comment]
will take a closer look

**Comment by john-co** [comment]
The testcase has `feature.flag.LPS-162766.type=beta` but you changed the macro to now use `release` virtual scope

**Comment by eltonccdantas** [comment]
> The testcase has `feature.flag.LPS-162766.type=beta` but you changed the macro to now use `release` virtual scope

Oh... ic...  weird 'cause it worked locally. I'll update it. Thanks John!

**Comment by eltonccdantas** [comment]
ci:test:peds-sm:nocompile

**Comment by eltonccdantas** [comment]
Hey @john-co, the failure on this pull request is not related to my changes, I'll be waiting on your review in order to forward it 🎉  🚀

**Comment by john-co** [comment]
Looks good to me! :sunglasses:
Let's clean up the number of commits.

**Comment by eltonccdantas** [comment]
ci:forward:force

**Pull request closed without merging**